extensions/omni_vfs/migrate/4_table_fs.sql

```diff
@@ -0,0 +1,111 @@
+create table table_fs_filesystems
+(
+    id   integer primary key generated always as identity,
+    name text not null unique
+);
+
+create table table_fs_files
+(
+    id            bigint primary key generated always as identity,
+    filesystem_id integer                not null references table_fs_filesystems (id),
+    filename      text[]                 not null,
```

Comment:
for `filename` column instead of storing components of path, how about just storing absolute canonicalized path otherwise it requires a lot of array spliting and joining. Prefix matching should be enough right which can be provided by a btree index